work_auth: Authentication

erastian opened the pull request:
- added methods in API module
- added Forms with styles (without type='file')
- added react-hook-form
- added yup as validator
- styles fixes
- added Button behavior as Link
- added missed icons in IconSet
- added docker
- finished login & register form with pages
- added axios interceptors
- added zustand
- created zustand store for session
- refactored sessionApi module
- added Auth & Guest guard

erastian committed:
work_auth: Authentication

- added login page
- added register page(WIP)
- added axios interceptors
- added methods in API module
- added Store
- added Context
- added Forms with styles (without type='file')
- added react-hook-form
- added yup as validator
- styles fixes
- added Button behavior as Link
- added missed icons in IconSet

mdziarkach commented on src/app/providers/index.tsx:
```diff
@@ -18,13 +20,17 @@ const queryClient = new QueryClient({
+      <BrowserRouter>
+        <QueryClientProvider client={ queryClient }>
+          <StoreProvider>
+            <AuthProvider>
```
you already use tanstack
are you sure that context providers still needed?
just example that I found https://dev.to/this-is-learning/react-query-authentication-flow-id2

mdziarkach commented on src/pages/login/Login.tsx:
```diff
@@ -0,0 +1,15 @@
+export function Login() {
+  useDocumentTitle('News Portal: Login');
+
+  const { Login } = useAuth();
```
You can use this line inside of LoginForm

erastian committed:
work_auth: Authentication

- added docker
- finished login & register form with pages
- added axios interceptors
- added zustand
- created zustand store for session
- refactored sessionApi module
- added Auth & Guest guard

mdziarkach commented on Dockerfile:
```diff
@@ -0,0 +1,18 @@
+ENV PATH /app/node_modules/.bin:$PATH
+COPY . /app
+RUN npm ci --loglevel verbose
+CMD ["npm", "run", "dev"]
```
for what do you need this file in auth PR?

mdziarkach commented on src/app/providers/RouterProvider.tsx:
```diff
@@ -4,6 +4,7 @@ import { PAGE_PATH } from "~shared/config";
 
 import { MainLayout } from "~pages/main-layout/MainLayout.tsx";
 import { AuthLayout } from "~pages/auth-layout/AuthLayout.tsx";
+// import { sessionModel } from "~entities/session";
```
remove it if you don't already need this

mdziarkach commented on src/app/providers/RouterProvider.tsx:
```diff
@@ -12,6 +13,8 @@ const ArticlePage = lazy(() => import('~pages/article-page'));
 const Page404 = lazy(() => import('~pages/page-404'));
 
 export function Router() {
+  // const isAuth = sessionModel.useAuth();
```
remove this line please

mdziarkach commented on src/entities/session/api/sessionApi.ts:
```diff
@@ -0,0 +1,28 @@
+import { useQuery } from "@tanstack/react-query";
```
you wanted to remove react-query
why is it here again?

mdziarkach replied:
![image](https://github.com/erastian/REACT_news_client/assets/15855781/7ae3a22c-58bd-489c-ae20-6f234a51708b)

erastian committed:
work_auth: Authentication

- removed zustand
- refactored axios code

erastian committed:
work_auth: Authentication

- auth finished
